### ISSUE TITLE
Commit documentation into VCS?

### ISSUE THREAD

**Issue by brollb** [NetsBlox/NetsBlox#3107]
Should we commit the documentation into git? These files appear to be missing in the docker containers currently which is not really great. 

Since `npm run postinstall` will rebuild them anyway, it might not be a bad idea to commit them. (This is how we handle building the client-side JS files.)

**Comment by dragazo** [comment]
You mean commit the generated rst/html itself? I guess we could do that, but it might be a lot of files. I thought the docker containers ran postinstall in their build instructions. Do the doc files need to be present before that?

**Comment by brollb** [comment]
Good question. I am not sure why they don't exist in the docker container; it looks like `postinstall` should be run when it is being built... Do you mind digging into this? Once we have this up on dev.netsblox.org, I was hoping to ping the others about it so we can get RoboScape docs up and start linking to it (after the next release but we can do that soonish).

**Comment by dragazo** [comment]
Sure, I'll look into that. Does dev use run in a docker container? Cause I see the docs up on [dev](https://dev.netsblox.org/docs/).

**Comment by brollb** [comment]
Yeah, I hopped into the docker container and built those manually while looking into why they weren't showing up on dev.netsblox.org. :)

**Comment by dragazo** [comment]
Oh ok, in that case it might just somehow be failing on build. That's a good place for me to start

**Comment by dragazo** [comment]
But you didn't need to install any other dependencies to do it, right? just ran postinstall manually?

**Comment by brollb** [comment]
Yep
________________________________

**Comment by dragazo** [comment]
I just built the image with this (fun) command: `sudo docker build -t netsblox/base:latest -f Dockerfile.base . && sudo docker build -t netsblox/server:latest . && sudo docker run -it -v $(pwd):/netsblox --network host netsblox/base:latest /bin/bash` and it looks like it successfully built the html and I could host it from within docker. Do you have the build logs for the one that failed?

**Comment by dragazo** [comment]
I got it to build successfully, but the files just don't seem to be there in the container. Shouldn't that also be a problem with minify? From googling, the only similar problem was when volumes were involved, but I ran docker inspect on the final layer/image and it has no volumes.

**Comment by brollb** [comment]
When we run `npm install`, it looks like npm is printing a warning and skipping the postinstall:
![DeepinScreenshot_select-area_20210504084227](https://user-images.githubusercontent.com/4982789/117012474-993dff80-acb4-11eb-8eaa-dc13f936b351.png)

It's also a bit weird that the earlier command is working for you because the postinstall process currently requires a connection to mongodb which shouldn't be available in the isolated container. How is that step passing for you? Did you add the `--network host` flag?

**Comment by brollb** [comment]
That said, it might actually be good to make it a prestart script rather than postinstall. Even if we fix the docker build, the documentation would still be static for the docker container. 

This is problematic though since the documentation depends on the loaded services which can depend upon the `ENV` environment variable. In other words, it would be nice for dev.netsblox.org/docs to contain docs for all the available services - including `@alpha` services. Anyway, since the available services are not actually known until the server actually starts (bc they can depend upon environment variables), I agree that they should be moved to being built on server start (probably as an `npm prestart` script).

**Comment by dragazo** [comment]
Yeah, I had the `--network host` flag. The way I got it to finally (almost) work is bad though:I actually added mongodb to the install dependencies and manually started it. I only did that to see if that was the issue, which it partially is. But another issue is that thing with files not persisting after build (not sure how minify does that, if it even does atm).

I like the prestart script idea. I think that'll bypass the problem. And the way we have it currently, it's already all separate functions in `build.js`, so I'll go ahead and pull it out quickly and open a draft PR with the changes.
